page-objects/pages/HMWRKHomeworkListPage.js

```diff
@@ -1,135 +1,117 @@
+                let nameOfTheTaskSelector = await driver.$(".col-xl-12 > li:nth-child(" + i + ") > .content > h2")
+                let nameOfTheTask = await nameOfTheTaskSelector.getText()
+                if (await nameOfTheTask.includes(taskname)) {
+                    return i
+                }
+            }
+        }
+        return 0
+    },
 
-	chooseTaskAmongAllTasks: async function (taskname) {
-		let taskindex = await this.returnTaskChildIndex(taskname);
-		if (taskindex > 0) {
-			let task = await driver.$(
-				".col-xl-12 > li:nth-child(" + taskindex + ") > a > span.more"
-			);
-			await task.click();
-			await driver.pause(1500);
-			let selectorToBeLoaded = await driver.$(selectors.pageTitleSelector);
-			await selectorToBeLoaded.waitForExist(2000);
-		} else {
-			console.log("No such task was found");
-		}
-	},
+    chooseTaskAmongAllTasks: async function (taskname) {
+        let taskindex = await this.returnTaskChildIndex(taskname)
+        if (taskindex > 0) {
+            let task = await driver.$(".col-xl-12 > li:nth-child(" + taskindex + ") > a > span.more")
+            await task.click()
+            await driver.pause(1500)
+            let selectorToBeLoaded = await driver.$(pageTitleSelector)
+            await selectorToBeLoaded.waitForExist(2000)
+        } else {
+            console.log("No such task was found")
+        }
+    },
 
-	verify: async function (taskname) {
-		await this.goToHomeworkListPage();
-		await this.sortHometasks();
-		await this.chooseTaskAmongAllTasks(taskname);
-		let pageTitleSelector = await driver.$(selectors.pageTitleSelector);
-		let courseAndTaskName = await pageTitleSelector.getText();
-		let tasknameArray = await courseAndTaskName.split("- ");
-		let foundtaskName = tasknameArray[1];
-		await expect(taskname).to.equal(foundtaskName);
-	},
+    verify: async function (taskname) {
+        await this.goToHomeworkListPage()
+        await this.sortHometasks()
+        await this.chooseTaskAmongAllTasks(taskname)
+        let pageTitle = await waitHelpers.waitUntilElementIsPresent(pageTitleSelector)
+        let courseAndTaskName = await pageTitle.getText()
+        let tasknameArray = await courseAndTaskName.split("- ")
+        let foundtaskName = tasknameArray[1]
+        await expect(taskname).to.equal(foundtaskName)
+    },
 
-	areThereAnyTasks: async function () {
-		let elementWithTasks = await driver.$$(selectors.elementWithTasks);
-		return elementWithTasks.length > 0 ? true : false;
-	},
+    areThereAnyTasks: async function () {
+        let listOfTasks = await driver.$$(taskElement)
+        return listOfTasks.length > 0 ? true : false
+    },
 
-	getAllTasks: async function () {
-		return await Promise.all(
-			(await driver.$$("h2.h6")).map(
-				async (element) => await element.getText()
-			)
-		);
-	},
+    getAllTasks: async function () {
+        return await Promise.all((await driver.$$("h2.h6")).map(async (element) => await element.getText()))
```

Comment:
$$("h2.h6") this selector should also be defined outside the function - can be done later